container/src/main/java/com/flipkart/poseidon/core/PoseidonServlet.java

```diff
@@ -148,20 +151,26 @@ protected void doRequest(HttpMethod method, HttpServletRequest httpRequest, Http
 
     private void setRequestContext(HttpServletRequest httpServletRequest) {
         RequestContext.set(METHOD, httpServletRequest.getMethod());
-        RequestContext.set(REQUEST_ID, getRequestId(httpServletRequest));
-        RequestContext.set(IS_PERF_TEST, isPerfTest(httpServletRequest));
```

Comment:
I have added a field in HeaderConfiguration, asking if the particular header is required in RequestContext